### PR TITLE
Fix leaking sublevel progress

### PR DESCRIPTION
**Pull request by cforkish**
currently, when we display the sublevels for a bubble choice level, the sublevels will show progress if the user has progress for that level _in any script_. this PR fixes that by including the script in the query for sublevel progress in `bubble_choice.rb`.

## Links

jira: [LP-1792]

[LP-1792]: https://codedotorg.atlassian.net/browse/LP-1792

### CODE REVIEW

File: dashboard/app/models/levels/bubble_choice.rb [hunk 1]
```diff
@@ -141,8 +141,16 @@ def summarize_sublevels(script_level: nil, user_id: nil, should_localize: false)
...
 
       if user_id
-        level_info[:perfect] = UserLevel.find_by(level: level, user_id: user_id)&.perfect?
-        level_info[:status] = level_info[:perfect] ? SharedConstants::LEVEL_STATUS.perfect : SharedConstants::LEVEL_STATUS.not_tried
+        level_info[:perfect] = UserLevel.find_by(
+          level: level,
+          script: script_level.try(:script),
+          user_id: user_id
+          )&.perfect?
+        level_info[:status] = if level_info[:perfect]
```
Comment (jamescodeorg) [inline dashboard/app/models/levels/bubble_choice.rb:149]: Mostly curious, why this change?  The ternary seems more canonical and readable?

**Review by jamescodeorg — APPROVED** [review]
Should we add a test to cover this case?

File: dashboard/app/models/levels/bubble_choice.rb (hunk 1 above)
Comment (cforkish) [reply dashboard/app/models/levels/bubble_choice.rb:149]: we have a [linting rule](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/MultilineTernaryOperator) that disallows multiline ternaries, and while we don't have a linting rule for line length, i still like to respect our style guide line length in any code i touch. i agree that the ternary seems more readable, but apparently that's not "the ruby way" to do it!

**Comment by cforkish** [comment]
i will add a test for this.

**Comment by cforkish** [comment]
@maureensturgeon after reading through [LP-1395] i doubt it's related to this, but i'll let you be the judge of that.

[LP-1395]: https://codedotorg.atlassian.net/browse/LP-1395
